store/file/square.go

```diff
@@ -0,0 +1,131 @@
+package file
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+type square [][]share.Share
+
+// readSquare reads Shares from the reader and returns a square. It assumes that the reader is
+// positioned at the beginning of the Shares. It knows the size of the Shares and the size of the
+// square, so reads from reader are limited to exactly the amount of data required.
+func readSquare(r io.Reader, shareSize, edsSize int) (square, error) {
+	odsLn := edsSize / 2
+
+	square := make(square, odsLn)
+	for i := range square {
+		square[i] = make([]share.Share, odsLn)
+		for j := range square[i] {
+			square[i][j] = make(share.Share, shareSize)
+		}
+	}
```

Comment:
Don't we want to read the square by Row number of reads instead of Row*Col reads? Not sure why you preferred reading share-by-share instead of row-by-row